packages/fxa-settings/src/pages/InlineTotpSetup/container.tsx

```diff
@@ -66,6 +69,58 @@ export const InlineTotpSetupContainer = ({
     [location, navigate]
   );
 
+  // Determine if the session is verified
+  useEffect(() => {
+    if (sessionVerified !== undefined) {
+      return;
+    }
+    (async () => {
+      // The user is navigated to this page by the web application in response to
+      // a sign-in attempt.  But let's do some sanity checks.
+      const verified = await session.isSessionVerified();
+      setSessionVerified(verified);
+    })();
+  }, [session, sessionVerified]);
+
+  // Determine if a totp needs to be setup, and if so trigger setup.
+  useEffect(() => {
+    if (totp !== undefined || totpStatus?.account.totp.verified === true) {
+      return;
+    }
+    (async () => {
+      const totpResp = await createTotp({
+        variables: {
+          input: {
+            metricsContext,
+          },
+        },
+      });
+      setTotp(totpResp.data?.createTotp);
+    })();
+  }, [createTotp, metricsContext, totpStatus, totpStatusLoading, totp]);
+
+  // Once state has settled, determine if user should be directed to another page
+  useEffect(() => {
+    if (!isSignedIn || !signinState) {
+      navTo('signup');
```

Comment:
If the user is not signed in, should they be directed to signup or hard navigate to index? Not certain in which scenario this might happen.

Reply:
Tested and see that signup handles the bounce back to index, WFM!